src/vmm/tests/integration_tests.rs

```diff
@@ -133,3 +135,88 @@ fn test_vmm_seccomp() {
         }
     }
 }
+
+#[test]
+fn test_dirty_bitmap_error() {
+    // Error case: dirty tracking disabled.
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            set_panic_hook();
+            let boot_source_cfg: BootSourceConfig =
+                MockBootSourceConfig::new().with_default_boot_args().into();
+            let resources: VmResources = MockVmResources::new()
+                .with_boot_source(boot_source_cfg)
+                .into();
+            let mut event_manager = EventManager::new().unwrap();
+            let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
+
+            let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
+            assert_eq!(
+                format!("{:?}", vmm.lock().unwrap().get_dirty_bitmap().err()),
+                "Some(DirtyBitmap(Error(2)))"
+            );
+            vmm.lock().unwrap().stop(0);
+        }
+        vmm_pid => {
+            // Parent process: wait for the vmm to exit.
+            let mut vmm_status: i32 = -1;
+            let pid_done = unsafe { libc::waitpid(vmm_pid, &mut vmm_status, 0) };
+            assert_eq!(pid_done, vmm_pid);
+            restore_stdin();
+            // If any panics occurred, its exit status will be != 0.
+            assert!(unsafe { libc::WIFEXITED(vmm_status) });
+            assert_eq!(unsafe { libc::WEXITSTATUS(vmm_status) }, 0);
+        }
+    }
+}
+
+#[test]
+#[cfg(target_arch = "x86_64")]
+fn test_dirty_bitmap_succes() {
+    // This test is `x86_64`-only until we come up with an `aarch64` kernel that dirties a lot
+    // of pages.
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
```

Comment:
I think this scenario becomes increasingly popular lately. Would it be possible to have a helper struct for handling this ?

Reply:
Yup, good idea, I logged https://github.com/firecracker-microvm/firecracker/issues/1887 to track it separately, this PR is already big and this is out of scope-ish.